jpype/_jarray.pyi

```diff
@@ -0,0 +1,59 @@
+# *****************************************************************************
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#   See NOTICE file for details.
+#
+# *****************************************************************************
+import typing
+
+
+__all__ = ['JArray']
+
+
+T = typing.TypeVar('T')
+
+
+class JArray(typing.Generic[T]):
+
+    def __new__(cls, tp: typing.Type[T], dims=1) -> JArray[T]:
+        ...
+
+    @classmethod
+    def of(cls, array, dtype=None):
```

Comment:
This could be typed I think?

Reply:
Unfortunately a type is considered an illegal value for `typing.Literal` so there isn't much point. It might make sense to add once the minimum supported version is 3.12 and `collections.abc.Buffer` can be used.

Reply:
Actually I should check if you can do `typing.Type[numpy.int8]` or whatever type in combination with typing.overload and see if it helps determine the output parameter type. If it does I'll add it. If not then I'll just say it didn't work. I won't get to it until tomorrow.

Reply:
Unfortunately overloads with `typing.Type[JBoolean]`, `typing.Type[JByte]` didn't have the wanted behavior and neither did `def of(cls, array, dtype: typing.Type[T]) -> JArray[T]:`. Unfortunately I think the best that can be done is to stick on the return type hint.

Reply:
I had an epiphany this morning and realized it would work the way I wanted if I used a different `typing.TypeVar`